gems/smithy/lib/smithy/plan.rb

```diff
@@ -3,18 +3,13 @@
 module Smithy
   # The Plan class is a simple data structure that holds the model, type, and options for a generator.
   class Plan
-    # @param model [Hash] The API model as a JSON hash.
-    # @param type [Symbol] The type of code to generate, either :client, :server, or :types.
-    # @param options [Hash] The options passed to the generator.
-    def initialize(model, type, options = {})
-      @model = Vise::Model.new(model)
```

Comment:
Yeah, I tried to make it work with model inside of plan, but the major issue I had was that model inside of plan could differ based on the welding step. For example in preprocess it was a hash and any other step it would have been a vise model.

Reply:
I also removed welding hooks out of initializer methods which I think is code smell.